site/docs/use-cases/1-LLM-pipeline/_sections/_usage_options/_chat_scenario.mdx

```diff
@@ -0,0 +1,230 @@
+                prompt = input('question:\n')
+            except EOFError:
+                break
+            answer = pipe.generate(prompt)
+            print('answer:\n')
+            print(answer)
+            print('\n----------\n')
+        # highlight-next-line
+        pipe.finish_chat()
+        ```
+    </TabItemPython>
+    <TabItemCpp>
+        ```cpp showLineNumbers
+        #include "openvino/genai/llm_pipeline.hpp"
+        #include <iostream>
+
+        int main(int argc, char* argv[]) {
+            std::string prompt;
+
+            std::string model_path = argv[1];
+            ov::genai::LLMPipeline pipe(model_path, "CPU");
+
+            ov::genai::GenerationConfig config;
+            config.max_new_tokens = 100;
+            config.num_beam_groups = 3;
+            config.num_beams = 15;
+            config.diversity_penalty = 1.0f;
+
+            // highlight-next-line
+            pipe.start_chat();
+            std::cout << "question:\n";
+            while (std::getline(std::cin, prompt)) {
+                std::cout << "answer:\n";
+                auto answer = pipe.generate(prompt, config);
+                std::cout << answer << std::endl;
+                std::cout << "\n----------\n"
+                    "question:\n";
+            }
+            // highlight-next-line
+            pipe.finish_chat();
+        }
+        ```
+    </TabItemCpp>
+</LanguageTabs>
+
+#### Streaming the Output
+
+For more interactive UIs during generation, you can stream output tokens.
+
+##### Streaming with Lambda Function
+In this example, a lambda function outputs words to the console immediately upon generation:
+
+<LanguageTabs>
+    <TabItemPython>
+        ```python showLineNumbers
+        import openvino_genai as ov_genai
+        pipe = ov_genai.LLMPipeline(model_path, "CPU")
+
+        # highlight-start
+        # Create a streamer lambda function
+        streamer = lambda x: print(x, end='', flush=True)
+        # highlight-end
+
+        # highlight-next-line
+        pipe.start_chat()
+        while True:
+            try:
+                prompt = input('question:\n')
+            except EOFError:
+                break
+            # highlight-next-line
+            pipe.generate(prompt, streamer=streamer, max_new_tokens=100)
+            print('\n----------\n')
+        # highlight-next-line
+        pipe.finish_chat()
+        ```
+    </TabItemPython>
+    <TabItemCpp>
+        ```cpp showLineNumbers
+        #include "openvino/genai/llm_pipeline.hpp"
+        #include <iostream>
+
+        int main(int argc, char* argv[]) {
+            std::string prompt;
+
+            std::string model_path = argv[1];
+            ov::genai::LLMPipeline pipe(model_path, "CPU");
+
+            // highlight-start
+            // Create a streamer lambda function
+            std::function<bool(std::string)> streamer = [](std::string word) {
+                std::cout << word << std::flush;
+                return false;
```

Comment:
A enum should be returned. Same for the class streamer

Reply:
Fixed